scripts/publishReleaseOnGithub.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+
+SCRIPT_LOCATION=$(dirname -- "$(readlink -f -- "${BASH_SOURCE[0]}")")
+# shellcheck source=./release_functions.sh
+source "${SCRIPT_LOCATION}/source_files/release_functions.sh"
+source "${SCRIPT_LOCATION}/source_files/helper_functions.sh"
+
+
+function create_message_body(){
+  verify_version_present_and_formatted $1 || return $?
+  version="$1"
+  release_tag="v$1"
+  is_draft=$(determine_is_draft "${RELEASE}")
+  json_adjacent_body="{
+    'tag_name' : '${release_tag}',
+    'name' : 'Release ${release_tag}',
+    'body' : 'Version ${version} release of JGiven.\nSource code may be found in Maven Central: https://search.maven.org/search?q=com.tngtech.jgiven',
+    'draft' : ${is_draft}
+  }"
+  replace_single_by_double_quotes "${json_adjacent_body}"
+  return $?
+}
+
+body=$(create_message_body "$1") || { code=$?;  echo "Failed to create message body with error ${code}"; exit ${code}; }
+action_token=$([ -n "$2" ] && echo "$2") || { echo "Failed to parse action token"; exit 1; }
```

Comment:
The script is hard to understand as I do not know what the second parameter is that is supposed to be handed to the script.

I think a better way of parsing input parsing ist something like this

while [[ -n "${1-}" ]] ; do
  case "${1}" in
    --old-version*)
      OLD_VERSION=${1#*=}
      ;;
    --new-version*)
      NEW_VERSION=${1#*=}
      ;;
    --repository-username*)
      SONATYPE_USERNAME=${1#*=}
      ;;
    --repository-password*)
      SONATYPE_PASSWORD=${1#*=}
      ;;
    --signing-key*)
      GPG_SIGNING_KEY=${1#*=}
      ;;
    --signing-password*)
      GPG_SIGNING_PASSWORD=${1#*=}
      ;;
    *)
      echo "Unknown option '${1}'"
      exit 1
      ;;
  esac
  shift
done

Have you considered something like that?

Reply:
Hmm, input processing based on options... Why not.